### DISCUSSION TITLE
Receiving classification probability vectors of each predicted bbox

### DISCUSSION THREAD

**Discussion by johanenforcer** [ultralytics/yolov5#2724] [Q&A]
What I am looking to do is to run each sample of a pool consisting of unlabeled samples through a forward pass in in order to look at their softmax probability vectors to then later use this information.  I assume in order to retrieve this information, I would have to call the detect.py with the chosen image files and store their probability vectors. I dont know whether my understanding is correct but does the x[:, 5:] in  x[:, 5:] *= x[:, 4:5], represent the probability vector assuming the first 5 columns are xywhc?

Furthermore, from what I can see would I have to manually change the detect function so that it is callable instead of having to run it through cmd, by for example just inputting the parser information as arguments in the function call?

Thanks in advance!

**Comment by glenn-jocher** [comment]
@johanenforcer yes that's correct. indices 5 onward are the class probabilities. 0-3 are the xywh box coordinates and 4 is the object confidence.

For inference you can either use detect.py from the command line (or called from another python function) or YOLOv5 PyTorch Hub models directly in your own python code. For the Hub option see the PyTorch Hub tutorial:

## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://github.com/ultralytics/yolov5/issues/2518)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [ONNX and TorchScript Export](https://github.com/ultralytics/yolov5/issues/251)
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Reply by johanenforcer** [reply]
I was also curious but say you want to use a model during training based on the pretrained models, is this still possible while using the pytorch hub so that I can use a seperate model than the pretrained ones?

**Reply by glenn-jocher** [reply]
@johanenforcer yes you can load any custom YOLOv5 model with PyTorch Hub. See PyTorch Hub tutorial for details.
